### PR TITLE
Better debugging for amp-bind, amp-list

### PR DESCRIPTION
**Pull request by choumx**
- Add `AMP.setLogLevel(<number>)` for embedded cases where adding `#log=<number>` is not possible.
- Improve `AMP.printState()` and support passing JSON expressions, e.g. `AMP.printState(foo.bar)`.
- Add a bunch of fine-grained dev logging to amp-bind and amp-list.

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -1294,21 +1304,21 @@ export class Bind {
 
   /**
    * Print out the current state in the console.
+   * @param {string=} opt_expression
```
Comment (alabiaga) [inline extensions/amp-bind/0.1/bind-impl.js]: just noting that optional parameters need not be prefixed with opt anymore though I actually prefer it.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -1294,21 +1304,21 @@ export class Bind {
...
-          return '[Circular]';
-        } else {
-          seen.push(value);
-        }
+  printState_(opt_expression) {
+    if (opt_expression) {
+      if (typeof opt_expression !== 'string') {
+        user().error(TAG, 'Invalid JSON expression. Example: ' +
+            'AMP.printState("foo.bar") to print current value of "foo.bar".');
+        return;
```
Comment (alabiaga) [inline extensions/amp-bind/0.1/bind-impl.js]: no need for return, it is never reached as an error is thrown. You actually commented on a similar change I made 😄

Comment (choumx) [reply extensions/amp-bind/0.1/bind-impl.js]: You need to return for `error()` but not for `assert()`. Only the latter throws a JS error.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 1 above)
Comment (choumx) [reply extensions/amp-bind/0.1/bind-impl.js]: AMP's (Malte's) style guide still uses them. `opt_` prefix is not necessary for params with default values though.

File: src/runtime.js [hunk 3]
```diff
@@ -212,13 +213,19 @@ function adoptShared(global, callback) {
...
    * @param {string} experimentId
    * @param {boolean=} opt_on
    * @return {boolean}
    */
   global.AMP.toggleExperiment = toggleExperiment.bind(null, global);
 
+  /**
+   * @param {!LogLevel} level
+   */
+  global.AMP.setLogLevel = overrideLogLevel.bind(null);
```
Comment (choumx) [inline src/runtime.js]: @jridgewell Any thoughts on this?

**Pull request merged**